### ISSUE TITLE
Setting class does not always hold the user's settings

### ISSUE THREAD

**Issue by mardy** [miurahr/aqtinstall#609]
**Describe the bug**
I'm working on some changes in aqt, and it seems to me that the `Settings` object is not really a singleton, but several instances exist of it. Only one of them has the user's settings.

**To Reproduce**
Add a line `print(f"Trusted mirrors from {self}")` in the implementation of the `trusted_mirrors` property in `helper.py`. The run aqt. You'll see that it prints different values for self. If you also have a configuration file with some custom settings, you'll see that they are often missing.

I'm trying to reimplement the SettingsClass as a singleton as described [here](https://stackoverflow.com/questions/6760685/creating-a-singleton-in-python) but I must be doing something wrong and so far I haven't had any luck in getting this work properly.

**Comment by mardy** [comment]
BTW, I'm running aqt like this:
```
python -m aqt -c ~/.config/aqt.ini install-qt windows desktop 6.2.0 win64_mingw81
```
I suspect that this might not be a real bug and only happen because of the way I'm running aqt (because if I directly modify the file under `site-packages` I don't see the same problem.

**Comment by ddalcino** [comment]
Is there any difference in the content contained in the different instances of Settings?

I had thought that the Settings object was only instantiated once, in the settings.py file, here: https://github.com/miurahr/aqtinstall/blob/04e02803bc1aea63ded70dd27dfc3d2c648a13fd/aqt/helper.py#L482

**Comment by mardy** [comment]
> Is there any difference in the content contained in the different instances of Settings?

Yes, and that's the problem. One instance has the user settings, on the other instances `load_settings()` is not called.

> I had thought that the Settings object was only instantiated once, in the settings.py file, here:
> 
> https://github.com/miurahr/aqtinstall/blob/04e02803bc1aea63ded70dd27dfc3d2c648a13fd/aqt/helper.py#L482

Yes, that's the code where they get instantiated. I'm confused, because code found in the root of a module should be executed only once, the first time that the module is imported. But this does not match what I'm seeing here. Where you able to reproduce the bug? I initially thought I had something wrong on my system, but now I've run many tests, and it appears that this code is executed more than once.

**Comment by ddalcino** [comment]
> Where you able to reproduce the bug?

No, but I’ll try when I’m home from work. This could have security implications, since the settings.ini is used to choose trusted mirrors.

In the past, Settings was implemented using the Borg pattern (multiple objects that all point to the same data). I was really impressed by that idea, and I did not understand why we moved away from Borg to the current pattern.

**Comment by mardy** [comment]
> No, but I’ll try when I’m home from work. This could have security implications, since the settings.ini is used to choose trusted mirrors.

I'm still debugging this. I added a line to print the thread id in helper.py:
```python
Settings = SettingsClass()
print("aqt: Instantiating new Settings in thread {}", threading.get_ident())
```
and I see that it prints a different thread ID. I wonder if it could be that the `requests` module creates threads, and that imports are re-executed in each thread?

**Comment by mardy** [comment]
I suspect that this is due to the `multiprocessing` package used in `installer.py`. This seems to be confimed by the fact that I don't see this problem when running the `list-qt` command, but only with `install-qt`.

**Comment by miurahr** [comment]
`SettingsClass` is not true singleton.

It would be better to take Borg design pattern.

**Comment by mardy** [comment]
Unfortunately this does not work, I still see the same problem in `aqt/archives.py`. I created [a branch](https://github.com/mardy/aqtinstall/tree/ignore-update-hash) where I'm adding a new option, and I see that inside the `QtArchives._download_update_xml()` method the Settings class returns the default value, not the real one.

**Comment by ddalcino** [comment]
@mardy Can you please provide more details? From the current master branch, I am unable to reproduce this problem.

I tried:
1. Debug logging: print the name of the `settings.ini` file being used to load Settings within the `installer()` function run in parallel, as well as some of the contents
2. Breakpoints in my IDE: I set a breakpoint in `QtArchives._download_update_xml()` and examined the `timeout` object, which comes from the settings file if not set with CLI args

In both cases, I am seeing the path to the settings file that I specified in CLI arguments, and the values specified in that file, not the defaults. As far as I can tell, this is fixed.

**Comment by ddalcino** [comment]
#613 was not part of the most recent release, 3.1.0. Maybe that's what you mean?

**Comment by mardy** [comment]
Hi @ddalcino! I'm using the master branch; if you have time, please try to fetch the branch mentioned in my previous comment, because then you should be able to verify that there is still one Settings instance not properly initialized.

**Comment by ddalcino** [comment]
> Hi @ddalcino! I'm using the master branch; if you have time, please try to fetch the branch mentioned in my previous comment, because then you should be able to verify that there is still one Settings instance not properly initialized.

I have checked out your branch, and it looks like the `ignore_updates_hash` is loading properly from the settings file. When I use `--config alt_settings_file.ini`, it loads the `ignore_updates_hash` value from that file, and it does not use the default `aqt/settings.ini` or the fallback value. This looks fixed to me; is there something I'm missing?
